audio/mods/protracker.cpp

```diff
@@ -101,15 +101,13 @@ class ProtrackerStream : public ::Audio::Paula {
 
 	void doPorta(int track) {
 		if (_track[track].portaToNote && _track[track].portaToNoteSpeed) {
-			if (_track[track].period < _track[track].portaToNote) {
-				_track[track].period += _track[track].portaToNoteSpeed;
-				if (_track[track].period > _track[track].portaToNote)
-					_track[track].period = _track[track].portaToNote;
-			} else if (_track[track].period > _track[track].portaToNote) {
-				_track[track].period -= _track[track].portaToNoteSpeed;
-				if (_track[track].period < _track[track].portaToNote)
-					_track[track].period = _track[track].portaToNote;
-			}
+			int distance = _track[track].period - _track[track].portaToNote;
```

Comment:
I have a small doubt now: with period and pirtaToNote being both uint16, do we get integral promotion to int before they are substracted, or is the result computed as unsigned before being casted to the int with a potential issue in the case where portaToNote is greater than period?

Reply:
They are promoted to int before the the subtraction.
http://en.cppreference.com/w/cpp/language/operator_arithmetic#Conversions